components/utils/src/links.rs

```diff
@@ -1,16 +1,8 @@
-pub fn anchor_id_checks(anchor:&str) -> Vec<String> {
-    vec![
-        format!(" id={}", anchor),
-        format!(" ID={}", anchor),
-        format!(" id='{}'", anchor),
-        format!(" ID='{}'", anchor),
-        format!(r#" id="{}""#, anchor),
-        format!(r#" ID="{}""#, anchor),
-        format!(" name={}", anchor),
-        format!(" NAME={}", anchor),
-        format!(" name='{}'", anchor),
-        format!(" NAME='{}'", anchor),
-        format!(r#" name="{}""#, anchor),
-        format!(r#" NAME="{}""#, anchor),
-    ]
+use regex::Regex;
+
+
+pub fn anchor_id_checks(anchor:&str) -> Regex {
+    Regex::new(
+        &format!(r#" (id|ID|name|NAME) *= *("|'){}("|')"#, anchor)
```

Comment:
you're missing the ones where they have no quotes

Reply:
I would also make regex case-insensitive to handle Name, iD, etc.

Reply:
Now done, hopefully, with any obscure side effects.